### PR TITLE
MINIFICPP-735: Fix issues with delimiter, make changes to facilitate …

### PR DESCRIPTION
**Pull request by phrocker**
…testing this change

Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by phrocker** [comment]
@jdye64 this should resolve your issues. Let travis run its course of course and feel free to verify.

File: libminifi/test/integration/IntegrationBase.h [hunk 1]
```diff
@@ -62,6 +62,10 @@ class IntegrationBase {
 
   }
 
+  virtual void updateProperties(std::shared_ptr<minifi::FlowController> fc) {
+
+    }
```
Comment (arpadboda) [inline libminifi/test/integration/IntegrationBase.h]: Please fix spacing / new lines here!

File: libminifi/src/processors/TailFile.cpp [hunk 2]
```diff
@@ -287,14 +288,36 @@ void TailFile::onTrigger(core::ProcessContext *context, core::ProcessSession *se
...
+              delim = '\t';
+              break;
+            case 'n':
+              delim = '\n';
+              break;
+            case '\\':
+              delim = '\\';
+              break;
+            default:
+              // previous behavior
```
Comment (arpadboda) [inline libminifi/src/processors/TailFile.cpp]: Maybe a log statement here?
This seems to be a configuration error as the delimiter string is longer than 1, but it doesn't contain any escaped char we support.

File: libminifi/test/integration/TailFileTest.cpp [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+#include <sys/stat.h>
+#undef NDEBUG
+#include <cassert>
+#include <utility>
+#include <chrono>
+#include <fstream>
+#include <memory>
+#include <string>
+#include <thread>
+#include <type_traits>
```
Comment (arpadboda) [inline libminifi/test/integration/TailFileTest.cpp]: Do we need all these?

File: libminifi/src/processors/TailFile.cpp (hunk 2 above)
Comment (phrocker) [reply libminifi/src/processors/TailFile.cpp]: That is the purpose of the log that was added below this.

**Comment by phrocker** [comment]
@jdye64 please review if you have a chance.

**Comment by jdye64** [comment]
I tested this out in several situations and this works as expected for me. Thanks @phrocker! Merge this bad boy in

**Comment by phrocker** [comment]
Merged under 54007b2ab7371bb8462ac46d89aa9969f3178d6b forgot closure tag.

**Pull request closed without merging**